File: my-folder/0048-rotate-image/solution.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        output = []
        rows = len(matrix)
        cols = len(matrix[0])
        for col in range(cols):
            for row in range(rows - 1, -1, -1):
                output.append(matrix[row][col])
        idx = 0
        for row in range(rows):
            for col in range(cols):
                matrix[row][col] = output[idx]
                idx += 1
        
        



        

File: my-folder/0048-rotate-image/solution.java
class Solution {
    public void rotate(int[][] matrix) {
        int top = 0;
        int bot = matrix.length - 1;


        while(top < bot){
            for(int i = 0; i < bot - top; i++){
                int left = top; 
                int right = bot;
                int first = matrix[top][left + i];
                matrix[top][left + i] = matrix[bot - i][left];
                matrix[bot - i][left] = matrix[bot][right - i];
                matrix[bot][right - i] = matrix[top + i][right];
                matrix[top + i][right] = first;
            }
            top++;
            bot--;
        }
    }
}


// 123 
// 456
// 789
// ----

// 00 copy this 
// update 00 with 20 
// 20 with 22 
// 22 with 02



